components/raftstore/src/coprocessor/region_info_accessor.rs

```diff
@@ -217,17 +242,43 @@ impl RoleObserver for RegionEventListener {
     }
 }
 
+impl RegionHeartbeatObserver for RegionEventListener {
+    fn on_region_heartbeat(&self, context: &mut ObserverContext<'_>, region_stat: &RegionStat) {
+        if !(self.region_stats_manager_enabled_cb)() {
+            // Region stats manager is disabled, return early.
+            return;
+        }
+        let region = context.region().clone();
+        let region_stat = region_stat.clone();
+        let event = RaftStoreEvent::UpdateRegionActivity {
+            region,
+            activity: RegionActivity { region_stat },
+        };
+
+        self.scheduler
+            .schedule(RegionInfoQuery::RaftStoreEvent(event))
```

Comment:
I'm afraid there will be a periodical message storm for the region-collector which may delay other important messages like "create region" and "region role change".

Reply:
Thanks for input, this has been merged a while ago, but since I am revisiting this area, I'll try to make this async or performance test to rule out this problem.